### ISSUE TITLE
Found 0 models (Possibly unable to override config for directories)

### ISSUE THREAD

**Issue by alariva** [beyondcode/laravel-er-diagram-generator#10]
Hi! Congrats and thanks for this pretty much convenient tool.

I've installed as instructed, but after running the command it seems to find no models. I've configured the proper directory and tried the text mode to help debug (As per this [issue](https://github.com/beyondcode/laravel-er-diagram-generator/issues/4#issuecomment-402388129)).

  * `PHP 7.2.7`
  * `laravel/framework v5.6.26`

```
$ php artisan generate:erd --format=text
Found 0 models.
Inspecting model relations.
digraph "G" {
style="filled"
bgcolor="#F7F7F7"
fontsize="12"
labelloc="t"
concentrate="1"
splines="polyline"
overlap=""
nodesep="1"
rankdir="LR"
pad="0.5"
ranksep="2"
esep="1"
fontname="Helvetica Neue"
}
```

```
// ./config/erd-generator.php
// ...
    'directories' => [
        app_path(),
    ],
// ...
```

I expect 8 models to be found:
```
$ ls --group-directories-first -1r app/
Validators
Traits
Providers
Policies
Patches
Observers
Notifications
Mail
Listeners
Http
FIMEDI
Exceptions
Events
Console
User.php
SaasQuota.php
Payment.php
Patient.php
Inspection.php
DeviceModel.php
Control.php
Account.php
```

Any possible clue is welcome. Thanks in advance!

**Comment by alariva** [comment]
```
    protected function getModelsThatShouldBeInspected(): Collection
    {
        $directories = config('erd-generator.directories');
dd($directories); // returns [], even with the correct config set up.
        $modelsFromDirectories = $this->getAllModelsFromEachDirectory($directories);
        return $modelsFromDirectories;
    }
```

Maybe there's something with `$this->mergeConfigFrom(__DIR__.'/../config/config.php', 'erd-generator');` in the service provider?

**Comment by alariva** [comment]
From https://laravel.com/docs/5.6/packages#configuration

> **(!)** This method only merges the first level of the configuration array. If your users partially define a multi-dimensional configuration array, the missing options will not be merged.

I'm getting vendor-level-config to override project-level-config values, and it should be the other way around. I'm confused, since this relies on the [long standing function mergeConfigFrom](https://github.com/laravel/framework/blame/5.6/src/Illuminate/Support/ServiceProvider.php#L55) and looks like everything should work as expected.

**Comment by mpociot** [comment]
This is strange. After you published your configuration, can you verify that your configuration is not cached?

**Comment by alariva** [comment]
It is. Config cache is a good point, but it isn't. Are you able to correctly override these settings?

```
$ head -20 config/erd-generator.php 
<?php

return [

    /*
     * All models in these directories will be scanned for ER diagram generation.
     * The directories will not be scanned recursively, so be sure to add all
     * directories that contain your models.
     */
    'directories' => [
        app_path(),
    ],

    /*
     * The generator will automatically try to look up the model specific columns
     * and add them to the generated output. If you do not wish to use this
     * feature, you can disable it here.
     */
    'use_db_schema' => true,
```
```
$ php artisan config:clear
Configuration cache cleared!
```
```
$ php artisan generate:erd --format=text

In Finder.php line 547:
                                                                   
  The "/home/alariva/fimedi/app/models" directory does not exist.  
```

**Comment by mpociot** [comment]
Can you retry the artisan command with `-vvv` to increase the verbosity?
I'm not using the Finder class, so I do not think that this package is the problem.

**Comment by EricEon** [comment]
I have to same problem. Found 0 models. @alariva , can u post the solution u found. Thanks

**Comment by alariva** [comment]
@mpociot 
```
$ php artisan generate:erd --format=text -vvv

In Finder.php line 547:
                                                                   
  [InvalidArgumentException]                                       
  The "/home/alariva/fimedi/app/models" directory does not exist.  
                                                                   

Exception trace:
 Symfony\Component\Finder\Finder->in() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php:390
 Illuminate\Filesystem\Filesystem->files() at /home/alariva/fimedi/vendor/beyondcode/laravel-er-diagram-generator/src/ModelFinder.php:26
 BeyondCode\ErdGenerator\ModelFinder->getModelsInDirectory() at /home/alariva/fimedi/vendor/beyondcode/laravel-er-diagram-generator/src/GenerateDiagramCommand.php:95
 BeyondCode\ErdGenerator\GenerateDiagramCommand->BeyondCode\ErdGenerator\{closure}() at n/a:n/a
 array_map() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Support/Collection.php:932
 Illuminate\Support\Collection->map() at /home/alariva/fimedi/vendor/beyondcode/laravel-er-diagram-generator/src/GenerateDiagramCommand.php:96
 BeyondCode\ErdGenerator\GenerateDiagramCommand->getAllModelsFromEachDirectory() at /home/alariva/fimedi/vendor/beyondcode/laravel-er-diagram-generator/src/GenerateDiagramCommand.php:85
 BeyondCode\ErdGenerator\GenerateDiagramCommand->getModelsThatShouldBeInspected() at /home/alariva/fimedi/vendor/beyondcode/laravel-er-diagram-generator/src/GenerateDiagramCommand.php:52
 BeyondCode\ErdGenerator\GenerateDiagramCommand->handle() at n/a:n/a
 call_user_func_array() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:29
 Illuminate\Container\BoundMethod::Illuminate\Container\{closure}() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:87
 Illuminate\Container\BoundMethod::callBoundMethod() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:31
 Illuminate\Container\BoundMethod::call() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Container/Container.php:564
 Illuminate\Container\Container->call() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Console/Command.php:184
 Illuminate\Console\Command->execute() at /home/alariva/fimedi/vendor/symfony/console/Command/Command.php:251
 Symfony\Component\Console\Command\Command->run() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Console/Command.php:171
 Illuminate\Console\Command->run() at /home/alariva/fimedi/vendor/symfony/console/Application.php:886
 Symfony\Component\Console\Application->doRunCommand() at /home/alariva/fimedi/vendor/symfony/console/Application.php:262
 Symfony\Component\Console\Application->doRun() at /home/alariva/fimedi/vendor/symfony/console/Application.php:145
 Symfony\Component\Console\Application->run() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Console/Application.php:89
 Illuminate\Console\Application->run() at /home/alariva/fimedi/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php:122
 Illuminate\Foundation\Console\Kernel->handle() at /home/alariva/fimedi/artisan:35
```

@EricEon Haven't found a solution just yet. A dirty workaround I got is altering the original config file in `vendor/` to get the desired `directories` value. Since it's a *dev* package it's ok for a one-time run, but of course this is not the right path for a perm-fix.

**Comment by alariva** [comment]
`laravel/framework v5.6.26`

When I run tinker the value *is correct*, but not when running the command, which makes everything more confusing.

```
$ php artisan tinker
Psy Shell v0.9.6 (PHP 7.2.7-1+ubuntu16.04.1+deb.sury.org+1 — cli) by Justin Hileman
>>> config('erd-generator');
=> [
     "directories" => [
       "/home/alariva/fimedi/app",
     ],
     "use_db_schema" => true,
     "table" => [
       "header_background_color" => "#d3d3d3",
       "header_font_color" => "#333333",
       "row_background_color" => "#ffffff",
       "row_font_color" => "#333333",
     ],
     "graph" => [
       "style" => "filled",
       "bgcolor" => "#F7F7F7",
       "fontsize" => 12,
       "labelloc" => "t",
       "concentrate" => true,
       "splines" => "polyline",
       "overlap" => false,
       "nodesep" => 1,
       "rankdir" => "LR",
       "pad" => 0.5,
       "ranksep" => 2,
       "esep" => true,
       "fontname" => "Helvetica Neue",
     ],
     "node" => [
       "margin" => 0,
       "shape" => "rectangle",
       "fontname" => "Helvetica Neue",
     ],
     "edge" => [
       "color" => "#003049",
       "penwidth" => 1.8,
       "fontname" => "Helvetica Neue",
     ],
     "relations" => [
       "HasOne" => [
         "dir" => "both",
         "color" => "#D62828",
         "arrowhead" => "tee",
         "arrowtail" => "none",
       ],
       "BelongsTo" => [
         "dir" => "both",
         "color" => "#F77F00",
         "arrowhead" => "tee",
         "arrowtail" => "crow",
       ],
       "HasMany" => [
         "dir" => "both",
         "color" => "#FCBF49",
         "arrowhead" => "crow",
         "arrowtail" => "none",
       ],
     ],
   ]
>>> 
```

However running the command still does not find any model:
```
$ php artisan config:clear
Configuration cache cleared!
$ php artisan generate:erd --format=text
Found 0 models.
Inspecting model relations.
digraph "G" {
style="filled"
bgcolor="#F7F7F7"
fontsize="12"
labelloc="t"
concentrate="1"
splines="polyline"
overlap=""
nodesep="1"
rankdir="LR"
pad="0.5"
ranksep="2"
esep="1"
fontname="Helvetica Neue"
}
```

**Comment by alariva** [comment]
Sorry, looks like it was my bad. I got it workig.

```
$ php artisan generate:erd --format=text
Found 8 models.
Inspecting model relations.
 8/8 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%digraph "G" {
style="filled"
bgcolor="#F7F7F7"
fontsize="12"
...
```

This is possibly what happened:
  * Possible issue with cached config (strange, it should not be this way, but maybe my fault here)
  * I altered some package files to debug the problem and tinker around, but looks like I broke everything up running after the incorrect bug
  * Cleared config cache just in case and as requested
  * Deleted the package dire and reinstalled via composer to make sure I had a frash install
  * Verified the config file at project level was correct
  * Running the command again loaded all models correctly

@mpociot  Thanks for the hints.
